packages/dev/query/src/cli.js

```diff
@@ -130,23 +131,113 @@ function getAsset(v: string) {
   }
 }
 
-function findAsset(v: string) {
+function _findAssetNode(v: string) {
   let assetRegex = new RegExp(v);
   for (let node of assetGraph.nodes.values()) {
     if (
       node.type === 'asset' &&
       assetRegex.test(fromProjectPathRelative(node.value.filePath))
     ) {
-      try {
-        console.log(
-          `${bundleGraph.getAssetPublicId(
-            bundleGraph.getAssetById(node.id),
-          )} ${fromProjectPathRelative(node.value.filePath)}`,
-        );
-      } catch (e) {
-        console.log(fromProjectPathRelative(node.value.filePath));
+      return node;
+    }
+  }
+}
+
+function findAsset(v: string) {
+  let node = _findAssetNode(v);
+  if (node) {
+    try {
+      console.log(
+        `${bundleGraph.getAssetPublicId(
+          bundleGraph.getAssetById(node.id),
+        )} ${fromProjectPathRelative(node.value.filePath)}`,
+      );
+    } catch (e) {
+      console.log(fromProjectPathRelative(node.value.filePath));
+    }
+  }
+}
+
+function findAssetWithSymbol(local: string) {
+  let [, assetId, binding, ref] = nullthrows(
+    local.match(/^\$([^$]+)\$([^$]+)\$(.*)$/),
+    `symbol ${local} could not be resolved`,
+  );
+
+  let asset;
+  // Search against the id used by the JSTransformer and ScopeHoistingPackager,
+  // not the final asset id, as it may have changed with further transformation.
+  for (let node of assetGraph.nodes.values()) {
+    if (node.type === 'asset' && node.value.meta.id === assetId) {
+      asset = node;
+      break;
+    }
+  }
+
+  // If the asset couldn't be found by searching for the id,
+  // search for the local name in asset used symbols.
+  if (asset == null) {
```

Comment:
Did you actually run into such a case? I can't think of a situation where the first search would fail

Reply:
I think you're right that this _shouldn't_ be possible, but if there is an undefined symbol at runtime (due to a bug in Parcel), it likely won't be discoverable via the asset graph, but _might_ still be discoverable in used symbols. (I have one such case that I am investigating and can share).

Reply:
Here's the case i mentioned: https://github.com/parcel-bundler/parcel/pull/8468